nucypher/policy/conditions/evm.py

```diff
@@ -0,0 +1,227 @@
+from nucypher.policy.conditions import STANDARD_ABIS
+from nucypher.policy.conditions._utils import CamelCaseSchema
+from nucypher.policy.conditions.base import ReencryptionCondition
+from nucypher.policy.conditions.lingo import ReturnValueTest
+
+_CONTEXT_DELIMITER = ':'
+_DIRECTIVES = {
+    'userAddress': lambda: True
+}
+CONTEXT_VARIABLES = {_CONTEXT_DELIMITER + k: v for k, v in _DIRECTIVES.items()}
+
+# TODO: Move this method to a util function
+__CHAINS = {
+    60: 'ethereum',  # TODO: make a few edits for now
+    61: 'testerchain',  # TODO: this one can be moved to a pytest fixture / setup logic
+    **PUBLIC_CHAINS,
+}
+
+
+def _resolve_chain(chain: Union[str, int]) -> Tuple[str, int]:
+    """Returns the name *and* chain ID given only a name *or* chain ID"""
+    for pair in __CHAINS.items():
+        if chain in pair:
+            chain_id, chain_name = pair
+            return chain_name, chain_id
+    else:
+        raise Exception(f'{chain} is not a known blockchain.')
+
+
+def _is_context_variable(parameter: str) -> bool:
+    return parameter.startswith(_CONTEXT_DELIMITER)
+
+
+def _is_valid_parameter(parameter: str) -> bool:
+    if _is_context_variable(parameter=parameter):
+        return parameter in CONTEXT_VARIABLES
+    else:
+        return True  # passthrough because this is not a context variable
+
+
+def _validate_parameters(parameters: List[Union[str, int]]):
+    for p in parameters:
+        if not _is_valid_parameter(parameter=p):
+            raise Exception(f'{p} is not a valid or authorized context variable')
+
+
+def _resolve_abi(standard_contract_type: str, method: str, function_abi: List) -> List:
+    """Resolves the contract an/or function ABI from a standard contract name"""
+    if not (function_abi or standard_contract_type):
+        # TODO: Is this protection needed?
+        raise ValueError(f'Ambiguous ABI - Supply either an ABI or a standard contract name.')
+    try:
+        function_abi = STANDARD_ABIS[standard_contract_type]
+    except KeyError:
+        if not function_abi:
+            raise Exception(f'No function ABI found')
+
+    # TODO: Verify that the function and ABI pair match?
+    # ABI(function_abi)
+    return function_abi
+
+
+def _process_parameters(parameters, **kwargs) -> List:
+    processed_parameters = []
+    for p in parameters:
+        if _is_context_variable(p):
+            real_name = p.strip(_CONTEXT_DELIMITER)
+            # TODO: camel case -> snake_case
+            # TODO: adhere to camelCase for the JSON standard or dive for easier integration here?
+            real_name = re.sub(r'(?<!^)(?=[A-Z])', '_', real_name).lower()
+            p = kwargs.get(real_name)
+            if not p:
+                raise Exception(f'"{real_name}" not found in request context')
+        processed_parameters.append(p)
+    return processed_parameters
+
+
+class RPCCondition(ReencryptionCondition):
+    ALLOWED_METHODS = (  # TODO: Deny list instead of allow list, if any?
```

Comment:
The only method I think we'd want to Deny is `[eth_sendRawTransaction](https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_sendrawtransaction)`